pydis_site/apps/api/tests/test_reminders.py

```diff
@@ -110,6 +111,42 @@ def test_delete_known_reminder_returns_204(self):
         self.assertRaises(Reminder.DoesNotExist, Reminder.objects.get, **{'id': self.reminder.id})
 
 
+class ReminderListTests(APISubdomainTestCase):
+    @classmethod
+    def setUpTestData(cls):
+        cls.author = User.objects.create(
+            id=6789,
+            name='Patrick Star',
+            discriminator=6789,
+            avatar_hash=None,
+        )
+
+        cls.reminder = Reminder.objects.create(
+            author=cls.author,
+            content="We should take Bikini Bottom, and push it somewhere else!",
+            expiration=datetime.utcnow().isoformat(),
+            jump_url="https://www.icantseemyforehead.com",
+            channel_id=123
+        )
+
+        cls.rem_dict = model_to_dict(cls.reminder)
+        cls.rem_dict['expiration'] += 'Z'  # Massaging a quirk of the response time format
+
+    def test_reminder_in_full_list(self):
+        url = reverse('bot:reminder-list', host='api')
+        response = self.client.get(url)
+
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.json(), [self.rem_dict])
+
+    def test_filter_search(self):
+        url = reverse('bot:reminder-list', host='api')
+        response = self.client.get(f'{url}?search={self.author.name}')
+
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.json(), [self.rem_dict])
+
+
```

Comment:
A test for filtering on fields is still missing. [Here](https://github.com/python-discord/site/blob/master/pydis_site/apps/api/tests/test_infractions.py#L90)'s an example. It may be more effective to have two reminder objects created so that a filter can be used to only return one of the two objects.

Reply:
Okay, I'm with you now (and this has helped me understand the `ReminderViewSet` code better, too).